packages/cdktf/lib/runtime.ts

```diff
@@ -24,6 +25,13 @@ export const booleanToTerraform: Mapper = identity;
 export const anyToTerraform: Mapper = identity;
 export const numberToTerraform: Mapper = identity;
 
+export function setMapper(elementMapper: Mapper): Mapper {
+  return (x: any) => {
+    if (!canInspect(x)) return x; // do not wrap undefined in toset()
+    return Fn.toset(listMapper(elementMapper)(x));
```

Comment:
Does it need to be wrapped in `toset` or will terraform contextually know that it's a set?

Reply:
I _think_ Terraform might imply it. I have to check that.

Reply:
I removed it 👍